src/selection/selectChild.js

```diff
@@ -0,0 +1,17 @@
+import matcher from "../matcher.js";
+import constant from "../constant.js";
+
+var find = Array.prototype.find;
+
+function childFind(match) {
+  match = typeof match === "function" ? match
+        : match == null ? constant(true)
+        : matcher(match);
+  return function() {
+    return find.call(this.children, function(e) { return match.call(e); });
```

Comment:
It feels weird to me that the *match* function isn’t passed any arguments, and that it only receives a child node as *this*. Normally when you pass a function to D3, it’s invoked for each selected element and passed the *datum*, *index*, and *group* as arguments, with *this* as the selected element.

I don’t think the normal D3 convention makes sense here because the function won’t be invoked once for each selected element: it’ll be invoked for each child of each selected element. So perhaps we should follow Array.prototype.find:

```suggestion
    return find.call(this.children, match);
```

We’d need to use something other than d3.matcher if *match* is not a function or nullish (i.e., a string), though.

```js
function childMatcher(selector) {
  return function(node) {
    return node.matches(selector);
  };
}
```